treeherder/webapp/api/performance_data.py

```diff
@@ -743,6 +748,254 @@ def _filter_out_retriggers(serialized_data: List[dict]) -> List[dict]:
         return serialized_data
 
 
+class PerfCompareResults(generics.ListAPIView):
+    serializer_class = PerfCompareResultsSerializer
+    queryset = None
+    noise_metric_header = 'noise metric'
+
+    def list(self, request):
+        query_params = PerfCompareResultsQueryParamsSerializer(data=request.query_params)
+        if not query_params.is_valid():
+            return Response(data=query_params.errors, status=HTTP_400_BAD_REQUEST)
+
+        base_revision = query_params.validated_data['base_revision']
+        new_revision = query_params.validated_data['new_revision']
+        base_repository_name = query_params.validated_data['base_repository']
+        new_repository_name = query_params.validated_data['new_repository']
+        interval = query_params.validated_data['interval']
+        framework = query_params.validated_data['framework']
+        no_subtests = query_params.validated_data['no_subtests']
+
+        base_signatures = PerformanceSignature.objects.select_related(
+            'framework', 'repository', 'platform', 'push', 'job'
+        ).filter(repository__name=base_repository_name)
+        new_signatures = PerformanceSignature.objects.select_related(
+            'framework', 'repository', 'platform', 'push', 'job'
+        ).filter(repository__name=new_repository_name)
+
+        base_signatures = base_signatures.filter(parent_signature__isnull=no_subtests)
+        new_signatures = new_signatures.filter(parent_signature__isnull=no_subtests)
+
+        if framework:
+            base_signatures = base_signatures.filter(framework__id=framework)
+            new_signatures = new_signatures.filter(framework__id=framework)
+
+        if interval:
+            base_signatures = base_signatures.filter(
+                last_updated__gte=datetime.datetime.utcfromtimestamp(
+                    int(time.time() - int(interval))
+                )
+            )
+            new_signatures = new_signatures.filter(
+                last_updated__gte=datetime.datetime.utcfromtimestamp(
+                    int(time.time() - int(interval))
+                )
+            )
+
+        base_signatures = base_signatures.values(
+            'framework_id',
+            'id',
+            'extra_options',
+            'suite',
+            'platform__platform',
+            'test',
+            'option_collection_id',
+            'repository_id',
+            'measurement_unit',
+        )
+        new_signatures = new_signatures.values(
+            'framework_id',
+            'id',
+            'extra_options',
+            'suite',
+            'platform__platform',
+            'test',
+            'option_collection_id',
+            'repository_id',
+            'measurement_unit',
+        )
+
+        base_signature_ids = [item['id'] for item in list(base_signatures)]
+        new_signature_ids = [item['id'] for item in list(new_signatures)]
```

Comment:
I see a lot of repitition for new and base, maybe for things like that you can extract to helper functions and just base in the required?